# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1440014946 14400
#      Wed Aug 19 16:09:06 2015 -0400
# Node ID 1a22083b59295db567987951c3ae0286c4595e96
# Parent  92ac2e05f393048a46e6ad8acff54f2dd4b05c18
use different approach to avoid unsigned/signed comparison warning

* txt-eng-ft.cc (ft_render::process_character): Cast value in
comparison instead of changing declaration.

diff --git a/libinterp/corefcn/txt-eng-ft.cc b/libinterp/corefcn/txt-eng-ft.cc
--- a/libinterp/corefcn/txt-eng-ft.cc
+++ b/libinterp/corefcn/txt-eng-ft.cc
@@ -555,18 +555,18 @@ ft_render::process_character (FT_ULong c
 
                   // 'w' seems to have a negative -1
                   // face->glyph->bitmap_left, this is so we don't
                   // index out of bound, and assumes we we allocated
                   // the right amount of horizontal space in the bbox.
                   if (x0 < 0)
                     x0 = 0;
 
-                  for (unsigned int r = 0; r < bitmap.rows; r++)
-                    for (unsigned int c = 0; c < bitmap.width; c++)
+                  for (int r = 0; static_cast<unsigned int> (r) < bitmap.rows; r++)
+                    for (int c = 0; static_cast<unsigned int> (c) < bitmap.width; c++)
                       {
                         unsigned char pix = bitmap.buffer[r*bitmap.width+c];
                         if (x0+c < 0 || x0+c >= pixels.dim2 ()
                             || y0-r < 0 || y0-r >= pixels.dim3 ())
                           {
                             //::warning ("ft_render: pixel out of bound (char=%d, (x,y)=(%d,%d), (w,h)=(%d,%d)",
                             //           str[i], x0+c, y0-r, pixels.dim2 (), pixels.dim3 ());
                           }
